ts/defs.bzl

```diff
@@ -172,6 +173,12 @@ def ts_project(
 
             See [docs/transpiler.md](/docs/transpiler.md) for more details.
 
+        declaration_transpiler: A custom transpiler tool to run that produces the TypeScript declaration outputs instead of `tsc`.
+
```

Comment:
Mention that if set, it should be a rule or macro with the signature `(name, srcs, **kwargs)`